engine/src/flutter/lib/ui/hooks.dart

```diff
@@ -417,6 +422,24 @@ void _invoke3<A1, A2, A3>(
   }
 }
 
+R? _invoke3WithReturn<A1, A2, A3, R>(
```

Comment:
can't think of a better name, open to suggestion.

Reply:
I think the name is inline with the existing invokes. Please add a doc comment, though